### ISSUE TITLE
Don't works on iOS 6.1

### ISSUE THREAD

**Issue by k06a** [coolstar/RecordMyScreen#19]
I have tried it on both iPhone 4 and iPhone 5.
App don't crashes, but it records nothing...

**Comment by coolstar** [comment]
Hmm, it works on my iPod 5 on 6.0.1, but when I update to 6.1 I'll check and see what's wrong.
Also, do you see anything in the log, and does it successfully record if you're in the app?
Also, try rebooting your phone and recording.

**Comment by k06a** [comment]
There are no warnings while recording.
5 warnings when trying to watch:

```
Feb  2 11:13:51 iPhone-k06a RecordMyScreen[4781] <Error>: CGContextSaveGState: invalid context 0x0
Feb  2 11:13:51 iPhone-k06a RecordMyScreen[4781] <Error>: CGContextClipToRect: invalid context 0x0
Feb  2 11:13:51 iPhone-k06a RecordMyScreen[4781] <Error>: CGContextTranslateCTM: invalid context 0x0
Feb  2 11:13:51 iPhone-k06a RecordMyScreen[4781] <Error>: CGContextDrawShading: invalid context 0x0
Feb  2 11:13:51 iPhone-k06a RecordMyScreen[4781] <Error>: CGContextRestoreGState: invalid context 0x0
```

**Comment by coolstar** [comment]
Hmm... that is odd indeed. I'll update to iOS 6.1 in a few days and will test it then.

**Comment by Bensge** [comment]
Yes video recording is broken on my iPhone 5 with 6.1 too

**Comment by coolstar** [comment]
_sigh_ I guess we have to wait for the jailbreak then :(

**Comment by loretoparisi** [comment]
I confirm the issue iPhone5/iOS 6.1.

The audio is being recorded correctly in the .caf file.
The video .mp4 file is corrupted instead.

**Comment by coolstar** [comment]
Ok, I've found what the issue is. captureShot: isn't even being called once for some reason, see for yourself by sticking an NSLog in there.

This means something other than the screenshot method changed in 6.1

**Comment by coolstar** [comment]
Ok, so the loop is entered but for some reason it doesn't enter this if statement: https://github.com/coolstar/RecordMyScreen/blob/master/RecordMyScreen/CSRecordViewController.m#L164

Update: It turns out that the long int diff, which is defined in https://github.com/coolstar/RecordMyScreen/blob/master/RecordMyScreen/CSRecordViewController.m#L161 gets -348516491 in iOS 6.1, while the int msBeforeNextCapture gets 41, which invalidates the if statement.

**Comment by coolstar** [comment]
RecordMyScreen has now been fixed for iOS 6.1 as of [a9e7d781bf6b8bfc4002cd172c8715d2531e326f](https://github.com/coolstar/RecordMyScreen/commit/a9e7d781bf6b8bfc4002cd172c8715d2531e326f)
